### DISCUSSION TITLE
UI ignores route prefix on redirect to home page.

### DISCUSSION THREAD

**Discussion by scott-david-walker** [prometheus/prometheus#12597] [Q&A]
### What did you do?

When you make use of the route prefix it gets ignored by the UI and redirects immediately to /graph page.

We have set external url to our hostname E.G. `https://myurl.com` and route prefix to `/sys/metrics` but when going to https://myurl.com/sys/metrics it redirects to /graph where nothing is available. Whereas if we go directly to https://myurl.com/sys/metrics/graph then everything works as expected.

We think it's an issue with the code below where the get route for the prefix might be overwritten by the next get route which is always applied

https://github.com/prometheus/prometheus/blob/94edd088595b3f872fd770418c8face76a853628/web/web.go#L354-L374

### What did you expect to see?

Going to a route with a prefix should redirect to `{prefix}/graph`

### What did you see instead? Under which circumstances?

Going to a route with a prefix redirects to `/graph` without prefix applied and therefore redirects to an invalid route

### Prometheus version

```text
version	2.41.0
revision	c0d8a56c69014279464c0e15d8bfb0e153af0dab
branch	HEAD
buildUser	root@d20a03e77067
buildDate	20221220-10:40:45
goVersion	go1.19.4
```

### Prometheus configuration file

```yaml
--web.route-prefix=/sys/metrics
--web.external-url=https://myurl.com
```

**Comment by roidelapluie** [comment]
It looks like you should use  --web.external-url=https://myurl.com/sys/metrics

**Reply by scott-david-walker** [reply]
I still don't feel that this is a valid response. We seem to be using the parameters in the correct way and the only thing not working is the redirect. It feels strange to have the option of using the prefix only for the response to be 'don't use the prefix'.

I do see this as a bug so prefer to have the issue reopened.

We're using this [helm chart](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus) and doing the proposed workaround causes issues finding the readiness and liveness endpoint.
